### PR TITLE
Patient match query params

### PR DESCRIPTION
**Pull request by elenarfowler**
Created new routes for find dup patients and index patient get with matches functions using query params and optional addr age and phone. Modified sql query to allow for empty addr, phone, and null age. Excluded deleted patients in sql query.

### CODE REVIEW

File: app/femr/data/daos/system/PatientRepository.java [hunk 1]
```diff
@@ -246,23 +246,32 @@ public IPatient retrievePatientById(Integer id) {
...
+            }
+
 
             String sql
-                    = "select id, user_id, first_name, last_name, phone_number, age, sex, address, city, isDeleted, deleted_by_user_id, reason_deleted, ("+
-                    "case when phone_number = " + phone + " then 40 else 0 end + " +
-                    "case when last_name = \"" + lastName +"\" then 15 else 0 end + " +
+                    = "select id, user_id, first_name, last_name, phone_number, age, sex, address, city, isDeleted, deleted_by_user_id, reason_deleted, (";
+            if(phone != null && !phone.equals("")) {
+                sql += "case when phone_number = " + phone + " then 40 else 0 end + ";
```
Comment (spencerklawans) [inline app/femr/data/daos/system/PatientRepository.java:258]: Is there a way to make the null phone check in raw SQL so it doesnt break up the statement like it does?

File: app/femr/data/daos/system/PatientRepository.java [hunk 2]
```diff
@@ -246,23 +246,32 @@ public IPatient retrievePatientById(Integer id) {
...
-                    "case when sex = \"" + gender + "\" then 10 else 0 end + " +
+                    "case when dm_first_name = dm(\"" + firstName +"\") then 10 else 0 end + ";
+            if(addr != null && !addr.equals("")) {
+                sql += "case when address = \"" + addr +"\" then 15 else 0 end + ";
+            }
+            if(age != null) {
+                sql += "case when age != null and age = \"" + ageString + "\" then 10 else 0 end + ";
+            }
+
+            sql += "case when sex = \"" + gender + "\" then 10 else 0 end + " +
```
Comment (spencerklawans) [inline app/femr/data/daos/system/PatientRepository.java]: Why are there two null checks for age here? The one in the if statement and the one in the sql line on 268.

File: app/femr/data/daos/system/PatientRepository.java (hunk 1 above)
Comment (spencerklawans) [inline app/femr/data/daos/system/PatientRepository.java]: Is there a way to make the null phone check in raw SQL so it doesnt break up the statement like it does?

**Review by spencerklawans — COMMENTED** [review]
Overall looks good!

File: app/femr/data/daos/system/PatientRepository.java (hunk 2 above)
Comment (elenarfowler) [reply app/femr/data/daos/system/PatientRepository.java]: The first check is for the param age to the function. The second checks if the current row being examined has age that is null before trying to actually compare the value.

**Review by elenarfowler — COMMENTED** [review]
I modified the sql query to include embedded if statements for checking null function params.

**Pull request merged**